### ISSUE TITLE
Editing the clusterSpec makes it complain about kubeletSpec resolvConf field not being recognized.

### ISSUE THREAD

**Issue by ottoyiu** [kubernetes/kops#2908]
With latest kops 1.7.0.alpha.1, when doing `kops get cluster --full -oyaml`, `resolvConf` underneath the `kubelet` section is dumped out as null. However, when editing the cluster spec and adding a `kubelet` section to it, it complains about the merged document not being valid:

```
# Found fields that are not recognized
# ...
#     kubelet:
#       allowPrivileged: true
# -     resolvConf: null
```

setting `resolvConf: ""` with an empty string, allows me to save the edit. However, it seems that the feature implemented here:
#2662 

would make it so that by specifying it as an empty string, it will invoke a different behaviour as desired.

@austinmoore- have you ran into something similar?

**Comment by chrislovecnm** [comment]
The problem is here:
https://github.com/kubernetes/kops/blob/master/pkg/apis/kops/componentconfig.go#L245

```go
ResolverConfig *string `json:"resolvConf" flag:"resolv-conf" flag-include-empty:"true"`
```
It was made `flag-include-empty:"true"` but not `omitempty`.

**Comment by austinmoore-** [comment]
Sorry I didn't catch this one. The unit tests only ensure that the `KubeletConfigSpec` struct gets correctly converted into flags. And when I tested this on my own, I just checked with only the `resolvConf` field set not any others. We need to add tests "higher" that test the YAML is generated correctly. I've opened a PR (#2916) with the fixes. I'm planning on getting the tests in with that PR as well, once I find a way to properly test this.
